ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,178 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let create =
+  call ~name:"create" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (String, "name_label", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ; (Bool, "enabled", "The enabled of the TracerProvider")
+      ; (Map (String, String), "tags", "The tags of the TracerProvider")
+      ; (Set String, "endpoints", "The endpoints of the TracerProvider")
+      ; (Set String, "components", "The componenets of the TracerProvider")
+      ; (Set String, "filters", "The filters of the TracerProvider")
+      ; (Set String, "processors", "The processors of the TracerProvider")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let destroy =
+  call ~name:"destroy" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:[(Ref _tracing, "self", "The TracerProvider to be destoryed")]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_hosts =
+  call ~name:"set_hosts" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the host field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_enabled =
+  call ~name:"set_enabled" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the enabled field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Bool
+        , "enabled"
+        , "The enabled the TracerProvider is to be set to. true meaning the \
+           TracerProvider is enabled and false that it is not"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_tags =
+  call ~name:"set_tags" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the tags field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Map (String, String)
+        , "tags"
+        , "The tags that are emitted to spans derived from this provider"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_endpoints =
+  call ~name:"set_endpoints" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the endpoints field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set String
+        , "endpoints"
+        , "The endpoints that the TracerProvider will export spans to. A URL \
```

Comment:
"List of endpoints (comma-separated) .. Each endpoint is either a URL or 'bugtool'"